src/lib/y2partitioner/actions/resize_partition.rb

```diff
@@ -0,0 +1,137 @@
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "yast/i18n"
+require "y2partitioner/dialogs/partition_resize"
+require "y2partitioner/ui_state"
+
+Yast.import "Popup"
+
+module Y2Partitioner
+  module Actions
+    # Action for resizing a partition
+    class ResizePartition
+      include Yast::I18n
+
+      # Constructor
+      #
+      # @param partition [Y2Storage::Partition]
+      def initialize(partition)
+        textdomain "storage"
+
+        @partition = partition
+        @resize_info = partition.detect_resize_info
+        UIState.instance.select_row(partition)
+      end
+
+      # Checks whether it is possible to resize the partition, and if so,
+      # the action is performed.
+      #
+      # @note An error popup is shown when the partition cannot be resized.
+      #
+      # @return [Symbol, nil]
+      def run
+        return :back unless validate
+        resize
+      end
+
+    private
+
+      # @return [Y2Storage::Partition] partition to resize
+      attr_reader :partition
+
+      # @return [Y2Storage::ResizeInfo]
+      attr_reader :resize_info
+
+      # Runs the dialog to resize the partition
+      #
+      # @return [Symbol] :finish if the dialog returns :next; dialog result otherwise.
+      def resize
+        result = Dialogs::PartitionResize.run(partition, resize_info)
+        fix_end_alignment(resize_info)
```

Comment:
Why is `resize_info` passed as argument if we have an attr_reader for it. It looks inconsistent to what we do with `partition`.